src/sidebar-counters/sidebar-counters.js

```diff
@@ -4,66 +4,80 @@
-    navLinksContainer.style.overflow = 'hidden'
+const HREF_BRANCHES = 'branches'
+const HREF_PULL_REQUESTS = 'pull-requests'
+export const menus = [HREF_BRANCHES, HREF_PULL_REQUESTS]
 
-    a.style.position = 'relative'
+const MAX_COUNTER = 99
 
-    const size =
-        // eslint-disable-next-line eqeqeq, no-eq-null
-        resources == null || resources.size == null
-            ? '?'
-            : resources.size > 999999
-                ? '999999+'
-                : resources.size
+export let menusCounters = {
+    HREF_BRANCHES: null,
+    HREF_PULL_REQUESTS: null,
+}
+
+type ResultSize = $Exact<{ size: number }>
 
-    const badge = (
+export function getBadge(size: number): HTMLElement {
+    const maxReached = size > MAX_COUNTER
+    return (
         <span class="__rbb-badge">
-            <span class="__rbb-badge-counter">{size}</span>
+            <span class={`__rbb-badge-counter${maxReached ? ' -max' : ''}`}>
+                {typeof size !== 'number' ? '!' : Math.min(size, MAX_COUNTER)}
+            </span>
         </span>
     )
-    a.insertBefore(badge, a.firstChild)
-    return badge
 }
 
-export default async function addSideBarCounters() {
-    // Exit early if can't find any of the nav links
-    // that we are going to augment with badge counters
-    const branchesLink: HTMLAnchorElement = await elementReady(
-        'a[href$="branches/"]'
-    )
-    const pullRequestsLink: HTMLAnchorElement = await elementReady(
-        'a[href$="pull-requests/"]'
-    )
-
-    if (!branchesLink || !pullRequestsLink) {
-        return
+async function getCounterInfo(wantedMenuFromHref: string): Promise {
+    switch (wantedMenuFromHref) {
+        case HREF_BRANCHES:
+            return api.getBranches()
+        case HREF_PULL_REQUESTS:
+            return api.getPullrequests()
+        default:
+            return
     }
+}
 
-    const [branches, pullrequests] = await Promise.all([
-        api.getBranches(),
-        api.getPullrequests(),
-    ])
+export async function addCounterToMenuItem(menu: HTMLElement) {
+    const link: HTMLAnchorElement = menu.querySelector('a')
 
-    const branchesBadge = addBadge(branchesLink, branches)
-    const prBadge = addBadge(pullRequestsLink, pullrequests)
+    if (!link) return
 
-    const resizeButton: HTMLButtonElement = await elementReady(
-        '.ak-navigation-resize-button'
+    const hrefParts = link.getAttribute('href').split('/')
+    const wantedMenuFromHref = hrefParts.find(x => menus.includes(x))
+
+    // Exit early if can't find one of the nav links
+    // that we are going to augment with badge counters
+    if (!wantedMenuFromHref) return
+
+    const size = menusCounters[wantedMenuFromHref]
+    const badge = getBadge(size)
+
+    menu.style.overflow = 'hidden'
+    link.style.position = 'relative'
+    link.insertBefore(badge, link.firstChild)
+}
+
+export default async function addSideBarCounters() {
+    menus.forEach(async menu => {
```

Comment:
getting the branches and prs counts when mounting the page so we will see more often a result badge, since the token is about 3 minutes long of live.

previously the selector was based on the <a href="/pullrequest" /> so for users like me that uses the contextual-floating-collapsed-menu, it only shows <a href on menu hover, so we need to have count data we before that action that may occur way after token expiration

Reply:
> getting the branches and prs counts when mounting the page so we will see more often a result badge, since the token is about 3 minutes long of live.

I don't understand what you mean here. In the previous implementation the count was also fetched at the same time you are fetching it here, on initial page load. I don't see how this results in the user seeing the result badge more often?

Reply:
yes you are right, this is call at the same time than the old behavior. What I meant is it will be more accurate just because the selector is different, both menus are binded to this function, even if they are hidden, it will get called once on mount

The old one was binded to the `<a href />` tag so if the page was loaded with the menu collapsed, it was not going to be called, resulting of a `?` if the user opened the menu after 3 minutes.